src/components/Tokens/TokenDetails/InvalidTokenDetails.tsx

```diff
@@ -69,9 +71,11 @@ export default function InvalidTokenDetails({
         </>
       ) : (
         <>
-          <InvalidDetailsText>
-            <Trans>This token doesn&apos;t exist on {getChainInfo(chainId)?.label}</Trans>
-          </InvalidDetailsText>
+          {connectedChainLabel && (
+            <InvalidDetailsText>
+              <Trans>This token doesn&apos;t exist on {connectedChainLabel}</Trans>
+            </InvalidDetailsText>
+          )}
           <TokenExploreButton onClick={() => selectChain(pageChainId)}>
```

Comment:
should this be hidden as well? Also reuse `connectedChainLabel` instead of re-calling `getChainInfo(pageChainId).label`

Reply:
this is actually a different chainId. on this page we can have `pageChainId` different from `connectedChainId`